### DISCUSSION TITLE
Как отложить выполнение действия и выполнить его в фоне

### DISCUSSION THREAD

**Discussion by erpic-ru** [totumonline/totum-ru-issues-and-discussions#161] [Как это сделать?] [answered]
Есть задача: постить ботом комментарии к определенной записи в телеграм-канале. С этим есть сложности, т.к. чтобы попасть в запись нужно знать id не отправленного сообщения, а id сообщения, которое сам телеграм пересылает в связанную группу (участники, которой могут комментировать записи).

В результате, после отправки сообщения нужно выждать некоторое время (желательно немного, т.к. потом сообщение может затеряться), затем загрузить последние записи (чем меньше, тем лучше), найти среди них нужную по id отправленного сообщения для дальнейших отправок комментариев к нему.

Если делать "в лоб", то телеграм не всегда успевает отдать обновление и приходит пустота. Я сделал через sleep, но это тормозит добавление строки и отправку основного сообщения (которое будет комментироваться).

Задача выглядит так: как запустить выполнение действия "фоном" через 5 секунд.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Идея такая:

1. Отделить это действие в отдельный процесс.

2. В этом отдельном процессе прописать таймер в n-секунд

Как это сделать. Делаем поле типа Текст:Тотум-код и пишем в него наш код действия с таймером.

```
=: while(action: $sleep; action: $action)

sleep: sleep(sec: 10)

action: ...
```

Теперь в том месте6 где й нас срабатывает триггер вызывающий это действие вызываем этот код через [exec](https://ru.docs.totum.online/functions#fn-exec) с параметром `ssh: true`.

`ssh: true` — как раз отделяет этот вызов в отдельный процесс и теперь основное действие продолжиться, а это подождет 10 сек и потом выполнится.

```
=: exec(code: @some_table.h_code_text; ssh: true)
```
